src/link.js

```diff
@@ -46,22 +46,72 @@ export default class Link extends Plugin {
 		this.editor.editing.view.addObserver( ClickObserver );
 
 		/**
-		 * Balloon panel view to display the main UI.
+		 * The form view inside {@link #balloonPanelView}.
 		 *
-		 * @member {module:link/ui/balloonpanel~BalloonPanelView}
+		 * @member {module:link/ui/linkformview~LinkFormView}
 		 */
-		this.balloonPanelView = this._createBalloonPanel();
+		this.formView = this._createForm();
 
 		/**
-		 * The form view inside {@link #balloonPanelView}.
+		 * Contextual balloon plugin instance.
 		 *
-		 * @member {module:link/ui/linkformview~LinkFormView}
+		 * @private
+		 * @member {module:ui/contextualballoon~ContextualBalloon}
 		 */
-		this.formView = this._createForm();
+		this._balloon = this.editor.plugins.get( ContextualBalloon );
 
 		// Create toolbar buttons.
 		this._createToolbarLinkButton();
 		this._createToolbarUnlinkButton();
+
+		// Attach lifecycle actions to the link balloon.
+		this._attachActions();
+	}
+
+	/**
+	 * Returns `true` when link view is added to the {@link module:ui/contextualballoon~ContextualBalloon}.
+	 *
+	 * @private
+	 * @returns {Boolean}
+	 */
+	get _isInBalloon() {
```

Comment:
For me `_isInBalloon()` is ok, but we can even drop this getter because it's only a syntax sugar for `this._balloon.hasView( this.formView )` and it's used only three times.